docs/metadata-versions.md

```diff
@@ -0,0 +1,23 @@
+# gittuf Metadata Versions
+
+Over time, gittuf metadata may change to support new features, for example. As
+existing metadata cannot be directly rewritten, a version number is added to
+signal to gittuf what to expect when parsing the underlying data.
+
+The version of the metadata can be found in the JSON property
+`metadata_version`, located in the root and targets metadata of the TUF metadata
+used in gittuf.
+
+Different versions of metadata do not necessarily imply breaking changes, but
+there may be such changes present (i.e. V0 -> V1).
+
+## Version 0 (Legacy)
```

Comment:
We have to think carefully of how this is communicated. For each version, I'd prefer to have schemas / migration descriptions so there's clarity on the changes.